File: app/src/main/java/com/arctouch/codechallenge/repository/SearchMoviesRepository.java
package com.arctouch.codechallenge.repository;

import com.arctouch.codechallenge.entity.UpcomingMoviesResponse;

import io.reactivex.Maybe;

public interface SearchMoviesRepository {
    Maybe<UpcomingMoviesResponse> getSearchMovies(String query);
    Maybe<UpcomingMoviesResponse> getSearchMovies(String query, int page);
}

File: app/src/main/java/com/arctouch/codechallenge/repository/SearchMoviesRepository.kt
package com.arctouch.codechallenge.repository

import com.arctouch.codechallenge.entity.UpcomingMoviesResponse

import io.reactivex.Maybe

interface SearchMoviesRepository {
    fun getSearchMovies(query: String): Maybe<UpcomingMoviesResponse>
    fun getSearchMovies(query: String, page: Int): Maybe<UpcomingMoviesResponse>
}
